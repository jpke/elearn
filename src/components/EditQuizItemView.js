import React from 'react';
import ListItems from './ListItems';

const EditQuizItemView = (props) => {

  let answers = props.answers.map((itemAnswer, idx) => {
    return (<div key={idx}>
              <input type="text" id={idx} name="itemAnswer" className="edit-quiz-item" value={itemAnswer.answer}/>
              <input type="radio" id={idx + "radio"} className="answer-radio" value={itemAnswer.correct} />
              <button className="deleteAnswer" onClick={(idx) => deleteAnswer.bind(this)}>Delete</button>
            </div>
    );
  });

  const deleteAnswer = (answerIndex) => {
    answers = answers.slice(0,answerIndex).concat(answers.slice(answerIndex+1,answers.length));
  };

  const addAnswer = (event) => {
    event.preventDefault();
    const newAnswer = {answer: "Add answer here", correct: false};
    answers.push(newAnswer);
  }

  return (
    <div className="quizInProgress">
      <h3>Question</h3>
      <p>{props.question}</p>
      <h3 className="answers">Answers</h3>
      <form onSubmit={props.editQuizItem} id="newQuizForm">
        {answers}
        <button className="addAnswer" onClick={addAnswer.bind(this)}>Add Answer</button>
        <button className="newQuizButton" type="submit">Update Question</button>
      </form>
    </div>
  );
};

export default EditQuizItemView;
