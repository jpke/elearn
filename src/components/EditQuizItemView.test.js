/* eslint-disable no-undef */
/* eslint-disable import/no-named-as-default */

import React from 'react';
import renderer from 'react-test-renderer';

import EditQuizItemView from './EditQuizItemView';

let question, answers;
beforeAll(() => {
  question = "no current question";
  answers =
    [
      {
        answer: "no current answer",
        correct: false
      },
      {
        answer: "no current answer true",
        correct: true
      }
    ];
})

it('renders correctly', () => {
  const tree = renderer.create(
    <EditQuizItemView
      key={1}
      question={question}
      answers={answers}
      index={1}
      editQuizItem={jest.fn()}
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
})
