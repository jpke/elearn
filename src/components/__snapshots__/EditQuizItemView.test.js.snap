exports[`test renders correctly 1`] = `
<div
  className="editQuiz-item">
  <h3>
    Question
  </h3>
  <div
    className="question-container">
    <textarea
      className="edit-question"
      id="1 question"
      name="itemQuestion"
      onChange={[Function]}
      type="text"
      value="no current question" />
  </div>
  <h3
    className="answers">
    Answers
  </h3>
  <div
    className="answer-container"
    id="1 answerDiv 0">
    <textarea
      className="edit-quiz-item"
      id="1 answer 0"
      name="itemAnswer"
      onChange={[Function]}
      type="text"
      value="no current answer" />
    <div
      className="buttons-container">
      <button
        className="modify-answers"
        id="1 answerCorrect 0"
        onClick={[Function]}>
        False
      </button>
      <button
        className="modify-answers"
        id="1 deleteAnswer 0"
        onClick={[Function]}>
        Delete
      </button>
    </div>
  </div>
  <div
    className="answer-container"
    id="1 answerDiv 1">
    <textarea
      className="edit-quiz-item"
      id="1 answer 1"
      name="itemAnswer"
      onChange={[Function]}
      type="text"
      value="no current answer true" />
    <div
      className="buttons-container">
      <button
        className="modify-answers"
        id="1 answerCorrect 1"
        onClick={[Function]}>
        True
      </button>
      <button
        className="modify-answers"
        id="1 deleteAnswer 1"
        onClick={[Function]}>
        Delete
      </button>
    </div>
  </div>
  <div
    className="item-buttons">
    <button
      className="modify-answers"
      id="1 addAnswer"
      onClick={[Function]}>
      Add Answer
    </button>
    <button
      className="modify-answers"
      id="1 deleteItem"
      onClick={[Function]}>
      Delete Question
    </button>
  </div>
</div>
`;
